appeals/web/src/client/components/add-another/add-another.js

```diff
@@ -51,17 +51,21 @@ function addRemoveButton(element) {
 	removeButton.classList.add(CLASSES.removeButton.additional);
 
 	const buttonContainer = element.querySelector(SELECTORS.removeButtonContainer) || element;
+	const buttonAlreadyExists = buttonContainer.querySelector('button');
 
-	buttonContainer.appendChild(removeButton);
+	if (!buttonAlreadyExists) {
+		buttonContainer.appendChild(removeButton);
+	}
 }
 
-function removeRemoveButtons(element) {
-	element.querySelectorAll(SELECTORS.removeButton).forEach((button) => button.remove());
+function removeRemoveButtons() {
+	document.querySelectorAll(SELECTORS.removeButton).forEach((button) => button.remove());
```

Comment:
Why was the `element` parameter removed? This component is designed to support multiple instances, which is why most of the functions in this file only perform queries like this in the scope of a parent element or a component instance - this will query the whole document, which could cause problems on pages where there are multiple instances of the "add another" component

Reply:
I think I see now, this is supposed to remove all remove buttons in the component - in that case you could just pass in `componentInstance` and change the `document.querySelectorAll` to `componentInstance.elements.root?.querySelectorAll`

Reply:
I've checked locally and this current code does cause a bug on pages with multiple instances of this component. For example, if you go to an appellant case, select "incomplete", check two checkboxes which show text fields, add an extra item to each of those, then click "remove" on one of them, it removes all the remove buttons from both sets of text fields